### ISSUE TITLE
Renaming is not case sensitive and is causing issues

### ISSUE THREAD

**Issue by ILoveSkilledCoders** [notepad-plus-plus/notepad-plus-plus#12396]
### Description of the Issue
By accident I had my caps on when I was renaming the file in NPP, & so when I went to rename it without the capped S to lower case, it appeared to work.

Then I restarted my computer & the capped S was back. This caused data issues.

Thank gawd I copied my data before closing the tab in question.

### Steps to Reproduce the Issue
See above

### Expected Behavior
The rename on NPP should work.

### Actual Behavior
Rename doesn't work.

### Debug Information
Notepad++ v8.4.6   (64-bit)
Build time : Sep 25 2022 - 19:51:39
Path : C:\Program Files\Notepad++\notepad++.exe
Command Line : "C:\Users\Laptop\AppData\Roaming\Notepad++\backup\TG Groups"
Admin mode : OFF
Local Conf mode : OFF
Cloud Config : OFF
OS Name : Windows 8.1 (64-bit) 
OS Build : 9600.20625
Current ANSI codepage : 1252
Plugins : 
    AutoSave (1.6.1)
    DSpellCheck (1.4.24)
    mimeTools (2.8)
    NppConverter (4.4)
    NppExport (0.4)

**Comment by alankilborn** [comment]
Casing of filenames is dependent upon Windows.  Typically Windows won't let you have two files that have names that differ only in case.

**Comment by alankilborn** [comment]
>(e.g. rename tab "Foo" to "FOO")

Ah, that's clearer.  Thank you.
I thought you were trying to rename a file-on-disk.  But really you were doing this:

![image](https://user-images.githubusercontent.com/77065706/197815024-eb25b8de-dcc7-419d-ae35-b545ee056794.png)

And you received this:

![image](https://user-images.githubusercontent.com/77065706/197815104-143f511f-d046-4df4-b2ac-be9d64d5bd8a.png)

It _somewhat_ makes sense.  As mentioned before, making case-insensitive-duplicate names is problematic at a Windows level.  Maybe this is some sort of vestige of that.

In my interpretation, it is a "bad thing" to attempt to do.
Can it be left as a "don't do that" or is it something you'd really like to see pursued and changed in Notepad++?

**Comment by ILoveSkilledCoders** [comment]
> Casing of filenames is dependent upon Windows. Typically Windows won't let you have two files that have names that differ only in case.

I didn't want 2 different file names, I just wanted to change the one file name b/c of my cap locks that had been on.

**Comment by xomx** [comment]
@ILoveSkilledCoders 
> Then I restarted my computer & the capped S was back. This caused data issues.

That is because if someone will try to rename a file under MS Windows, but change only upper/lowercase letters, the real low-level filename on disk often remains unchanged (it depends on more factors).

What I usually do in this situation is renaming the relevant file, changing the case of its characters to a desired form + adding 1 more character to its start or end, then renaming back again, this time with removing of that extra character:
- an invalid filename saved on disk: "iNVERTEDcASE.TXT"
- I rename to: "InvertedCase.txt#"
- then again to: "InvertedCase.txt"

**Comment by vahida1157** [comment]
@xomx 
> * an invalid filename saved on disk: "iNVERTEDcASE.TXT"
> * I rename to: "InvertedCase.txt#"
> * then again to: "InvertedCase.txt"

Yes, it's a solution we are using for now, but this is not user-friendly.
So, I think it would be good to check if the non-case-sensitive of file name didn't change, change the file and don't throw an error.

**Comment by xomx** [comment]
The N++ inherent case insensitivity has been fixed by the https://github.com/notepad-plus-plus/notepad-plus-plus/commit/2531b4d12c4eddc6c0a519eb37c6edf25196619e . This fix will be included in the upcoming v8.6.6. If you want to test it right now, you can try the corresponding GitHub Actions [build](https://github.com/notepad-plus-plus/notepad-plus-plus/actions/runs/8788038060)  ([HowTo](https://github.com/notepad-plus-plus/notepad-plus-plus/issues/14973#issuecomment-2049666226)).

The fix above will help with renaming of the unsaved N++ tabs (now one can rename e.g. "new 1" tab to "NEW 1" and there will not be anymore the "Rename failed" dlg visible in the above @alankilborn [post](https://github.com/notepad-plus-plus/notepad-plus-plus/issues/12396#issuecomment-1290746929)). But it will not really help with the renaming of the already existing files on the disk (the aforementioned Windows feature/limitation...).

@vahida1157 
If you mean that N++ should inherently use that existing-file-renaming-hack mentioned by me - I think that it should be doable by using the MoveFileEx WINAPI, rather than directly follow this double-renaming hack. But that needs some testing 1st.
